Zephyr 3.2 Upgrade (#1499)
zmkfirmware/zmk: docs/docs/features/bluetooth.md

## Profiles

By default, ZMK supports five "profiles" for selecting which bonded host
device should receive the keyboard input.

:::note Connection Management

When pairing to a host device ZMK saves bond information to the selected profile. It will not replace this automatically when you initiate pairing with another device. To pair with a new device select an unused profile with or clearing the current profile, using the [`&bt` behavior](../behaviors/bluetooth.md) on your keyboard.

A ZMK device may show as "connected" on multiple hosts at the same time. This is working as intended, and only the host associated with the active profile will receive keystrokes.

:::

Failure to manage the profiles can result in unexpected/broken behavior with hosts due to bond key mismatches, so it is an important aspect of ZMK to understand.

## Bluetooth Behavior

> Management of the bluetooth in ZMK is accomplished using the [bluetooth behavior](../behaviors/bluetooth.md). Be sure to refer to that documentation to learn how to manage profiles, switch between connected hosts, etc.

Reviewer: Minor inconsistency: used "&bt behavior" above and "bluetooth behavior" here.

Author: Fixed. Thanks.

## Troubleshooting

## Known Issues

There are a few known issues related to BLE and ZMK:

### Windows Battery Reporting

> There is a known issue with Windows failing to update the battery information after connecting to a ZMK keyboard. You can work around this Windows bug with overriding a [Bluetooth config variable](../config/bluetooth.md).

Reviewer: Might be helpful to list the variable that needs to be set, since that page will likely have more than one variable on it in the future.

Author: Added.  Think that's enough?

## Profiles

By default, ZMK supports five "profiles" for selecting which bonded host
device should receive the keyboard input.

:::note Connection Management

When pairing to a host device ZMK saves bond information to the selected profile. It will not replace this automatically when you initiate pairing with another device. To pair with a new device select an unused profile with or clearing the current profile, using the [`&bt` behavior](../behaviors/bluetooth.md) on your keyboard.

A ZMK device may show as "connected" on multiple hosts at the same time. This is working as intended, and only the host associated with the active profile will receive keystrokes.

:::

Failure to manage the profiles can result in unexpected/broken behavior with hosts due to bond key mismatches, so it is an important aspect of ZMK to understand.

## Bluetooth Behavior

Management of the bluetooth in ZMK is accomplished using the [`&bt` behavior](../behaviors/bluetooth.md). Be sure to refer to that documentation to learn how to manage profiles, switch between connected hosts, etc.

## Troubleshooting

## Known Issues

There are a few known issues related to BLE and ZMK:

### Windows Battery Reporting

> There is a known issue with Windows failing to update the battery information after connecting to a ZMK keyboard. You can work around this Windows bug with overriding a [Bluetooth config variable](../config/bluetooth.md) to force battery notifications even for host that neglect to subscribe to them:

Reviewer 2: ```suggestion
There is a known issue with Windows failing to update the battery information after connecting to a ZMK keyboard. You can work around this Windows bug by overriding a [Bluetooth config variable](../config/bluetooth.md) to force battery notifications even if a host neglects to subscribe to them:
```

```
CONFIG_BT_GATT_ENFORCE_SUBSCRIPTION=n
```

### macOS Connected But Not Working

If you attempt to pair a ZMK keyboard from macOS in a way that causes a bonding issue, macOS may report the keyboard as connected, but fail to actually work. If this occurs:

1. Remove the keyboard from macOS using the Bluetooth control panel.

> 1. Invoke `&bt BT_CLR` on the keyboard by pressing the correct keys for your particular keymap.

Reviewer 2: ```suggestion
1. Invoke `&bt BT_CLR` on the keyboard while the profile associated with the macOS device is active, by pressing the correct keys for your particular keymap.
```
